processing/src/main/java/org/apache/druid/query/groupby/GroupingEngine.java

```diff
@@ -686,8 +686,11 @@ public Sequence<ResultRow> processSubtotalsSpec(
           processingConfig.intermediateComputeSizeBytes()
       );
 
-      List<String> queryDimNames = baseSubtotalQuery.getDimensions().stream().map(DimensionSpec::getOutputName)
-                                                    .collect(Collectors.toList());
+      List<String> queryDimNamesInOrder = baseSubtotalQuery.getOrderingAndDimensions(false)
+                                                           .getDimensions()
+                                                           .stream()
+                                                           .map(DimensionSpec::getOutputName)
+                                                           .collect(Collectors.toList());
```

Comment:
note: can this be hidden somewhere - like a method named: `getDimentsionColumnNames` or something